src/components/SensorRaw/index.js

```diff
@@ -9,15 +10,20 @@ import AirIcon from '@mui/icons-material/Air';
 import { blue, purple, red } from '@mui/material/colors';
 import useMessage from '../../hooks/useMessage';
 
-const SensorRaw = ({ }) => {
+
+const SensorGraphs = () => {
   const { message } = useMessage(PLUGIN_KEYS.SENSOR_RAW, '/sensor_raw', 500);
-  const { setOnline, setOffline, setData } = usePluginState(
-    PLUGIN_KEYS.SENSOR_RAW
-  );
+  const { setOnline, setOffline } = usePluginState(PLUGIN_KEYS.SENSOR_RAW);
+  const [data, setData] = useState([]);
 
   useEffect(() => {
-    setData({ timestamp: message?.header?.stamp?.secs });
-    message ? setOnline() : setOffline();
+    if (message) {
+      setData(prevData => [...prevData, { temp: message?.temperature.toFixed(2), humidity: message?.humidity.toFixed(3), pressure: message?.pressure.toFixed(3) }]);
```

Comment:
We should limit the size of this data array. It might be 120 as you only show that much data point in line chart.

Reply:
Also why did we set the data in the local state, but not in redux store?